pydatastructs/trees/tests/test_binary_trees.py

```diff
@@ -436,3 +436,109 @@ def test_SplayTree():
     assert str(s) == ("[(1, 2000, 2000, None), (None, 1000, 1000, None)]")
     assert str(t) == ("[(None, 100, 100, None), '', (6, 200, 200, None), (4, 50, 50, None), "
                       "(5, 30, 30, None), (None, 20, 20, None), (3, 55, 55, 0), '', '', '']")
+
+def test_RedBlackTree():
+    tree = RedBlackTree()
+    tree.insert(10,10)
+    tree.insert(18,18)
+    tree.insert(7,7)
+    tree.insert(15,15)
+    tree.insert(16,16)
+    tree.insert(30,30)
+    tree.insert(25,25)
+    tree.insert(40,40)
+    tree.insert(60,60)
+    tree.insert(2,2)
+    tree.insert(17,17)
+    tree.insert(6,6)
+
+    assert str(tree) == \
+        ("[(11, 10, 10, 3), (10, 18, 18, None), (None, 7, 7, None), (None, 15, 15, None),"
+        " (0, 16, 16, 6), (None, 30, 30, None), (1, 25, 25, 7), (5, 40, 40, 8),"
+        " (None, 60, 60, None), (None, 2, 2, None), (None, 17, 17, None), (9, 6, 6, 2)]")
+
+    tree = RedBlackTree()
+    tree.insert(10)
+    tree.insert(20)
+    tree.insert(30)
+    tree.insert(40)
+    tree.insert(50)
+    tree.insert(60)
+    tree.insert(70)
+    tree.insert(80)
+    tree.insert(90)
+    tree.insert(100)
+    tree.insert(110)
+    tree.insert(120)
+    tree.insert(130)
+    tree.insert(140)
+    tree.insert(150)
+    tree.insert(160)
+    tree.insert(170)
+    tree.insert(180)
+
+    assert tree._get_sibling(7) is None
+
+    assert str(tree) == \
+        ("[(None, 10, None, None), (0, 20, None, 2), (None, 30, None, None), (1, 40, None, 5),"
+        " (None, 50, None, None), (4, 60, None, 6), (None, 70, None, None), (3, 80, None, 11),"
+        " (None, 90, None, None), (8, 100, None, 10), (None, 110, None, None), (9, 120, None, 13),"
+        " (None, 130, None, None), (12, 140, None, 15), (None, 150, None, None), (14, 160, None, 16),"
+        " (None, 170, None, 17), (None, 180, None, None)]")
+
+    tree.delete(180)
+    tree.delete(130)
+    tree.delete(110)
+    tree.delete(190)
+    assert str(tree) == \
+        ("[(None, 10, None, None), (0, 20, None, 2), (None, 30, None, None), (1, 40, None, 5), (None, 50, None, None),"
+         " (4, 60, None, 6), (None, 70, None, None), (3, 80, None, 11), (None, 90, None, None), (8, 100, None, None), '',"
+         " (9, 120, None, 15), '', (None, 140, None, 14), (None, 150, None, None), (13, 160, None, 16), (None, 170, None, None)]")
+
+    tree.delete(170)
+    tree.delete(100)
+    tree.delete(60)
+    assert str(tree) == \
+        ("[(None, 10, None, None), (0, 20, None, 2), (None, 30, None, None), (1, 40, None, 5), (None, 50, None, None), (4, 70, None, None),"
```

Comment:
Right Child of `40` should be 50 but according to this array it is, 70.
Visualise your tree here, https://www.cs.usfca.edu/~galles/visualization/RedBlack.html

Reply:
The visualizer uses predecessor for replacing the node and therefore according to it right child of 40 is 50.
But we have used successor and hence getting `70 `as the right child of `40`.

Reply:
Could you use predecessor? It would be easy to compare. I hope it needs a small change in the code and it shouldn't affect other test cases.

Reply:
Okay. Lemme change it to the predecessor.